File: frontend/src/lib/permissions.ts
// Keep this list in lock-step with backend Permissions.cs. Drift here only affects
// compile-time validation of `<Can permission="...">`; runtime auth always uses the
// backend's claims.
export const PERMISSIONS = [
  'customers.view',
  'customers.create',
  'customers.update',
  'customers.delete',
  'products.view',
  'products.create',
  'products.update',
  'products.delete',
  'quotations.view',
  'quotations.create',
  'quotations.update',
  'quotations.delete',
  'quotations.print',
  'quotations.cancel_confirmed',
  'quotations.view_cost',
  'quotations.view_all',
  'quotations.transfer_own',
  'quotations.transfer_any',
  'quotations.clone_orphan',
  'quotations.bypass_lock',
  'quotations.accounting_confirm',
  'quotations.cancel_accounting_confirmed',
  'users.view',
  'users.create',
  'users.update',
  'users.delete',
  'roles.view',
  'roles.manage',
  'user_settings.manage',
  'system.manage_settings',
  'reports.revenue',
  'reports.profit',
  'reports.debt',
  'reports.delivery',
] as const;

export type Permission = (typeof PERMISSIONS)[number];

export const ROLES = ['ADMIN', 'SALES', 'WAREHOUSE', 'ACCOUNTANT', 'MANAGER'] as const;
export type Role = (typeof ROLES)[number];

File: backend/src/OrderMgmt.Domain/Constants/Permissions.cs
namespace OrderMgmt.Domain.Constants;

public static class Permissions
{
    public const string SystemModule = "system";
    public const string CatalogModule = "catalog";
    public const string SalesModule = "sales";
    public const string ReportModule = "report";

    public static class Users
    {
        public const string View = "users.view";
        public const string Create = "users.create";
        public const string Update = "users.update";
        public const string Delete = "users.delete";
    }

    public static class Roles
    {
        public const string View = "roles.view";
        public const string Manage = "roles.manage";
    }

    public static class Customers
    {
        public const string View = "customers.view";
        public const string Create = "customers.create";
        public const string Update = "customers.update";
        public const string Delete = "customers.delete";
    }

    public static class Products
    {
        public const string View = "products.view";
        public const string Create = "products.create";
        public const string Update = "products.update";
        public const string Delete = "products.delete";
    }

    public static class Quotations
    {
        public const string View = "quotations.view";
        public const string Create = "quotations.create";
        public const string Update = "quotations.update";
        public const string Delete = "quotations.delete";
        public const string Print = "quotations.print";
        public const string CancelConfirmed = "quotations.cancel_confirmed";
        public const string ViewCost = "quotations.view_cost";
        public const string ViewAll = "quotations.view_all";
        public const string TransferOwn = "quotations.transfer_own";
        public const string TransferAny = "quotations.transfer_any";
        public const string CloneOrphan = "quotations.clone_orphan";
        public const string BypassLock = "quotations.bypass_lock";
    }

    public static class UserSettings
    {
        public const string Manage = "user_settings.manage";
    }

    public static class Reports
    {
        public const string Revenue = "reports.revenue";
        public const string Profit = "reports.profit";
        public const string Debt = "reports.debt";
        public const string Delivery = "reports.delivery";
    }
}

public static class RoleCodes
{
    public const string Admin = "ADMIN";
    public const string Sales = "SALES";
    public const string Accountant = "ACCOUNTANT";
    public const string Warehouse = "WAREHOUSE";
    public const string Manager = "MANAGER";
}
